src/diracx/core/job_status.py

```diff
@@ -0,0 +1,148 @@
+from datetime import datetime, timezone
+from unittest.mock import MagicMock
+
+from sqlalchemy.exc import NoResultFound
+
+from diracx.core.models import (
+    JobStatusUpdate,
+    ScalarSearchOperator,
+    SetJobStatusReturn,
+)
+from diracx.core.utils import JobStatus
+from diracx.db.jobs.db import JobDB, JobLoggingDB
+
+
+async def set_job_status(
```

Comment:
This should be moved to `diracx.dbs` or `diracx.routers` as it creates a circular dependnecy between `diracx.core` and `diracx.dbs`.

The submodule's dependnecies should strictly follow something like:

```
diracx.core -> diracx.cli
            -> diracx.api
            -> diracx.dbs -> diracx.routers
                          -> diracx.tasks

diracx.client -> diracx.api
              -> diracx.cli

diracx.api -> diracx.cli
```